packages/svelte/src/internal/client/runtime.js

```diff
@@ -476,7 +476,7 @@ export function update_reaction(reaction) {
 		// we need to increment the read version to ensure that
 		// any dependencies in this reaction aren't marked with
 		// the same version
-		if (previous_reaction !== null) {
+		if (previous_reaction !== null && previous_reaction !== reaction) {
```

Comment:
Can we simplify this then?
```suggestion
		if (previous_reaction !== reaction) {
```

Reply:
If the `previous_reaction` is `null` then it would also apply it though, no?

Reply:
From a quick glance `reaction` cannot be falsy at this position, "not `null`" is included in that comparison